### PR TITLE
Document jvm options for cli tools

### LINKED ISSUE #92503: Document CLI_JAVA_OPTS
All CLI tools run with 64MB mem by default. For elasticsearch-node, this can be way too little, since it need to work on the cluster state. However, there is no documentation on how to change that.

We should document that `CLI_JAVA_OPTS` can be used to change java options for CLI tools.

### PR DESCRIPTION
**Pull request by rjernst**
This commit adds documentation for the CLI_JAVA_OPTS environment variable.

closes #92503

### CODE REVIEW

File: docs/reference/commands/node-tool.asciidoc [hunk 1]
```diff
@@ -50,6 +50,10 @@ This tool has a number of modes:
   {es}. This may sometimes allow you to downgrade to an earlier version of
   {es}.
 
+:tool-name node
```
Comment (nik9000) [inline docs/reference/commands/node-tool.asciidoc:53]: s/:tool-name node/:tool-name: node/

File: docs/reference/commands/cli-jvm-options.asciidoc [hunk 2]
```diff
@@ -0,0 +1,12 @@
+[[cli-tool-jvm-options]]
+=== JVM options for CLI tools
```
Comment (nik9000) [inline docs/reference/commands/cli-jvm-options.asciidoc:2]: Add a `=` here to make it a sub-section.

You'll want to include `{tool-name}` in the anchor above or we'll end up with duplicate anchors and those have to be unique across the whole project.

**Comment by rjernst** [comment]
@nik9000 I think this is ready for review now. I've added it to a few tools that I think make sense. Thanks for the help!

**Review by nik9000 — APPROVED** [review]
That was hard fought.

**Pull request merged**